src/components/common/TmConnectedNetwork.vue

```diff
@@ -68,19 +68,59 @@
         Connecting…
       </div>
     </div>
+    <hr v-if="currentNetwork.powered" class="powered-hr" />
+    <div v-if="currentNetwork.powered" class="powered-div">
+      <span class="powered-by">Powered by&nbsp;</span>
+      <Avatar
```

Comment:
does the avatar component handle the generation of the image if there is no picture?

the v-else-if below should be inside the avatar component, no?

Reply:
no, if there is not picture or picture is null then will go to the <img> element. But you are right the in img it could be just `v-else` actually, because there is no other condition

Reply:
so this smells like the validator image could be its own component - if theres an image, show the image, otherwise show the generated avatar. a task for another day.